server/schedulers/balance_leader.go

```diff
@@ -240,7 +244,18 @@ func (l *balanceLeaderScheduler) transferLeaderIn(cluster opt.Cluster, target *c
 		schedulerCounter.WithLabelValues(l.GetName(), "no-leader").Inc()
 		return nil
 	}
-	return l.createOperator(cluster, region, source, target)
+	targets := []*core.StoreInfo{
+		target,
+	}
+	finalFilters := l.filters
+	if leaderFilter := filter.NewPlacementLeaderSafeguard(l.GetName(), cluster, region, source); leaderFilter != nil {
+		finalFilters = append(l.filters, leaderFilter)
+	}
+	targets = filter.SelectTargetStores(targets, finalFilters, cluster)
+	if len(targets) < 1 {
+		return nil
```

Comment:
do we need to add a new inc in `schedulerCounter.WithLabelValues`

Reply:
good catch, updated.